src/Microsoft.Azure.WebJobs.Host/Common/ITriggerBindingStrategy.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// Strategy pattern to describe how to bind a core trigger type to various parameter.
+    /// Suports:
+    /// - both Single-item vs. Batch dispatch.  
+    /// - core binding, string, poco w/ Binding Contracts
+    /// 
+    /// For example, a single EventHubTriggerInput -->  can bind to 
+    ///  EventData, EventData[], string, string[], Poco, Poco[]    
+    /// </summary>
+    /// <typeparam name="TMessage">The native message type. For Azure Queues, this would be CloudQueueMessage.</typeparam>
+    /// <typeparam name="TTriggerValue">The type of the trigger object that the listener returns. This could represent a single message or a batch of messages.</typeparam>
+    public interface ITriggerBindingStrategy<TMessage, TTriggerValue>
+    {
+        /// <summary>
+        /// Given a raw string, convert to a TTriggerValue.
+        /// This is primarily used in the "invoke from dashboard" path. 
+        /// </summary>
+        /// <param name="message"></param>
+        /// <returns></returns>
+        TTriggerValue ConvertFromString(string message);
+
+        /// <summary>
+        /// Get the static route-parameter contract for the TMessage. 
+        /// For example, if we bind a queue message to a POCO, 
+        /// then the properties on the Poco's type are route parameters that can feed into other bindings. 
+        /// Intentionally make this mutable so that callers can add more items to it and override defaults. 
+        /// </summary>
+        /// <returns>A map of the names of items in the binding contract and their types.</returns>
+        Dictionary<string, Type> GetStaticBindingContract();
+
+        /// <summary>
+        /// Get the values of the route-parameters given an instance of the trigger value. 
+        /// This should match the strucutre in GetStaticBindingContract. 
+        /// Intentionally make this mutable so that callers can add more items to it. 
+        /// </summary>
+        /// <param name="value">The instance of the trigger object.</param>
+        /// <returns>Runtime values based on the trigger value for the binding contract.</returns>
+        Dictionary<string, object> GetContractInstance(TTriggerValue value);
+
+        /// <summary>
+        /// Bind as a single-item dispatch. 
+        /// This gets called when somebody has a function like:
+        ///     MyTrigger([Trigger] TMessage item)
+        /// </summary>
+        /// <param name="value">the trigger value</param>
+        /// <param name="context">runtime binding information</param>
+        /// <returns>the single message instance to bind to the user's function</returns>
+        TMessage BindMessage(TTriggerValue value, ValueBindingContext context);
```

Comment:
Recommend renaming to just "Bind" or "BindSingle". For example you could have Bind/BindArray, or Bind/BindMultiple